### PR TITLE
feat(helm): Add flag to force pod recreation as part of helm upgrade

### LINKED ISSUE #4897: Configuration changes don't restart Kubernetes pod
#### Environment

- OS: emqx/emqx Docker container
- Erlang/OTP: R22/10.7.2.6
- EMQ: 4.2.8

#### Description

If the environment variable configuration changes and thus the `emqx-env` configmap, the pod will not be restarted and the changes will not be applied. This can be tested by installing with `--set "emqxConfig.EMQX_ALLOW_ANONYMOUS=true"`. Upgrading the helm chart afterwards with `--set "emqxConfig.EMQX_ALLOW_ANONYMOUS=false"` will modify the `emqx-env` configmap, but not restart the pod.

With this issue it is hard to apply configuration changes via helm upgrades.

### PR DESCRIPTION
**Pull request by nicklasfrahm**
Fixes #4897.

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

## Backward Compatibility

Yes, this feature is disabled by default.

## More information

This has been tested manually on a kubernetes cluster by installing the helm chart and toggling `emqxConfig.EMQX_ALLOW_ANONYMOUS=true` between upgrades. To see if the configuration change was applied, an MQTT client was used to connect to the service.

### CODE REVIEW

File: deploy/charts/emqx/templates/StatefulSet.yaml [hunk 1]
```diff
@@ -47,6 +47,10 @@ spec:
         version: {{ .Chart.AppVersion }}
         app.kubernetes.io/name: {{ include "emqx.name" . }}
         app.kubernetes.io/instance: {{ .Release.Name }}
+      {{- if .Values.recreatePods }}
+      annotations:
+        recreated-pods-at: {{ now | quote }}
```
Comment (zmstone) [inline deploy/charts/emqx/templates/StatefulSet.yaml:52]: make sense to annotate with configmap hash instead ?
e.g.
`checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}`

ref: https://v2.helm.sh/docs/developing_charts/#automatically-roll-deployments-when-configmaps-or-secrets-change

Comment (nicklasfrahm) [reply deploy/charts/emqx/templates/StatefulSet.yaml]: @zmstone I fixed this and additionally also took care of the warnings for Kubernetes `>=1.17`.

**Comment by zmstone** [comment]
Thank you for the PR @nicklasfrahm 
@Rory-Z Please help to review it.

**Comment by nicklasfrahm** [comment]
@zmstone I can also bump the version. Due to the size of this PR, this feels like a patch version increment to me, so `4.3.2 > 4.3.3`. Do you agree?

**Comment by zmstone** [comment]
> @zmstone I can also bump the version. Due to the size of this PR, this feels like a patch version increment to me, so `4.3.2 > 4.3.3`. Do you agree?

Yes, agree.

**Comment by nicklasfrahm** [comment]
@zmstone Could you approve the CI? 🚀

File: src/emqx.app.src [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {application, emqx,
  [{id, "emqx"},
   {description, "EMQ X"},
-  {vsn, "4.3.2"}, % strict semver, bump manually!
```
Comment (zmstone) [inline src/emqx.app.src:4]: sorry, not here.
it should be `include/emqx_release.hrl`

Comment (zmstone) [reply src/emqx.app.src:4]: I also sent a PR to make it more clear: https://github.com/emqx/emqx/pull/4899

Comment (nicklasfrahm) [reply src/emqx.app.src]: @zmstone Could you kick off the CI?

File: include/emqx_release.hrl [hunk 3]
```diff
@@ -29,7 +29,7 @@
 
 -ifndef(EMQX_ENTERPRISE).
 
--define(EMQX_RELEASE, {opensource, "4.3.2"}).
```
Comment (Rory-Z) [inline include/emqx_release.hrl:32]: Hi, @nicklasfrahm 
Thank you for your contribution, but I think this part of the modification should be revoked, and we will update this flag before releasing the new version

**Review by Rory-Z — APPROVED** [review]

**Pull request merged**